hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/CacheConfig.java

```diff
@@ -398,6 +398,9 @@ public long getCacheCompactedBlocksOnWriteThreshold() {
    * configuration.
    */
   public boolean shouldReadBlockFromCache(BlockType blockType) {
+    if (!getBlockCache().isPresent()) {
```

Comment:
[nit] do we hit any bug, why does need to check blockcache instance exist after this change?